### PR TITLE
Try to publish unscoped packages too

### PR DESCRIPTION
**Pull request by Schniz**
This is a part of the movement towards scoped packages in all of Wix, to support a gradual conversion of packages.
tl;dr: right now, when we publish `package-a`, we also publish `@wix/package-a`. Let's do the opposite too: `@wix/package-a -> package-a`. We should be careful, so `@wix/draft-js` won't be published internally as `draft-js`.
This way, we could convert infra packages first.

## 🌮 Current Behavior

* package name (in `package.json`) is `package-a`
  * Wix internal registry will publish:
    * `package-a`
    * `@wix/package-a`
  * Public npm registry will publish:
    * `@wix/package-a`
* package name (in `package.json`) is `@wix/package-a`
  * Wix internal registry will publish:
    * `@wix/package-a`
  * Public npm registry will publish:
    * `@wix/package-a`

## 🤝 New, expected behavior:

* package name (in `package.json`) is `package-a`
  * *stays the same as before*
* package name (in `package.json`) is `@wix/package-a`
  * Wix internal registry will publish:
    * `package-a`
    * `@wix/package-a`
  * Public npm registry will publish:
    * `@wix/package-a`
* a package that doesn't have wix internal registry as the publish config registry (like `@wix/draft-js`, `yoshi`):
  * Wix internal registry will publish:
    * `@wix/draft-js`
  * Public npm registry will publish:
    * `@wix/draft-js`

### CODE REVIEW

**Comment by Schniz** [comment]
cc @yairhaimo @mastertheblaster

File: src/publish-scoped.js [hunk 1]
```diff
@@ -73,6 +95,8 @@ export function publishScoped() {
       publishToRegistry(pkg, 'http://npm.dev.wixpress.com/');
       if (!isScoped(bkp.name)) {
         publishToRegistry(pkg, 'https://registry.npmjs.org/');
+      } else {
+        publishUnscopedPackage(bkp);
```
Comment (mastertheblaster) [inline src/publish-scoped.js]: Maybe you would also like to update the package-lock files if they do exist.
`updateLockFiles(...name...)`

File: src/publish-scoped.js [hunk 2]
```diff
@@ -60,6 +64,24 @@ function updateLockFiles(packageName) {
   updateLockFileIfExists('package-lock.json', packageName);
 }
 
+function verifyWixPackage(packageName) {
+  const result = execSync(`npm view ${packageName} publishConfig.registry`).toString();
```
Comment (mastertheblaster) [inline src/publish-scoped.js]: What happens if package is not found or registry is down currently? Will it return or break?

Comment (mastertheblaster) [reply src/publish-scoped.js]: This one is expensive command:

```
time npm view edit-contact publishConfig.registry
http://repo.dev.wix/artifactory/api/npm/npm-local/
npm view edit-contact publishConfig.registry  0.49s user 0.07s system 3% cpu 15.883 total
```

took about `15s` to complete

**Comment by mastertheblaster** [comment]
Great job, but maybe you would like some safety trigger to quickly disable this feature? For example using some ENV variable.

File: src/publish-scoped.js (hunk 1 above)
Comment (mastertheblaster) [reply src/publish-scoped.js]:
```js
if (process.env.PUBLISH_UNSCOPED === 'true') {
   // ...
}
```
and afterwards we could ask CI guys to enable this env variable for some POC projects and if everything would go smoothly we could enable this flag for all CI.

File: src/publish-scoped.js (hunk 2 above)
Comment (Schniz) [reply src/publish-scoped.js]:
```
✡ ◦ time npm view edit-contact@latest publishConfig.registry
http://repo.dev.wix/artifactory/api/npm/npm-local/
npm view edit-contact@latest publishConfig.registry  0.44s user 0.07s system 9% cpu 5.321 total

✡ ◦ time (curl --silent http://repo.dev.wix/artifactory/api/npm/npm-local/edit-contact/latest | jq '.publishConfig.registry')
"http://repo.dev.wix/artifactory/api/npm/npm-local/"
( curl --silent  | jq '.publishConfig.registry'; )  0.04s user 0.01s system 14% cpu 0.393 total
```

the second one (direct `curl`) is _alot_ faster. Not sure if this should be used though.. wdyt?

Comment (mastertheblaster) [reply src/publish-scoped.js]: `Exactly like what happens when publishing` - this is bad IMHO,
cause package `packageName` might not exist and `execSync` will throw an error breaking the build.

File: src/publish-scoped.js (hunk 1 above)
Comment (Schniz) [reply src/publish-scoped.js]: why would I want to update lockfiles for unscoped packages?

File: src/publish-scoped.js (hunk 2 above)
Comment (Schniz) [reply src/publish-scoped.js]: I'm pushing some fixes, and here's the output for `verifyWixPackage`:

```javascript
> verifyWixPackage('this-package-surely-doesnt-exist-mantas!')
false
```

also, it shouldn't fail on unknown package, because the way I wrote it now 😄

File: src/publish-scoped.js [hunk 3]
```diff
@@ -60,6 +64,37 @@ function updateLockFiles(packageName) {
   updateLockFileIfExists('package-lock.json', packageName);
 }
 
+function fetchJSONSync(url) {
+  const result = spawnSync('curl', ['--silent', url], {encoding: 'utf8'}).stdout;
+  return JSON.parse(result);
+}
+
+function verifyWixPackage(packageName) {
```
Comment (Schniz) [inline src/publish-scoped.js]:
```javascript
> verifyWixPackage('this-package-surely-doesnt-exist-mantas!')
false

> verifyWixPackage('react')
false

> verifyWixPackage('santa-core-utils')
true
```

**Comment by Schniz** [comment]
Thanks for this awesome review, @mastertheblaster 😸 

I still have an open question: why should I update lockfiles when I publish the unscoped packages?

File: src/publish-scoped.js (hunk 3 above)
Comment (mastertheblaster) [reply src/publish-scoped.js]: 👍

File: src/publish-scoped.js [hunk 4]
```diff
@@ -60,6 +64,37 @@ function updateLockFiles(packageName) {
...
+    const packageContents = fetchJSONSync(`http://npm.dev.wixpress.com/${packageName}/latest`);
+    const result = packageContents.publishConfig && packageContents.publishConfig.registry;
+    return Boolean(
+      result && (
+        result.includes('npm.dev.wixpress.com') ||
+        result.match(/https?:\/\/repo.dev.wix\//)
+      )
+    );
+  } catch (e) {
+    console.error(`An error occured while looking for ${packageName} in Wix's registry:`, e);
```
Comment (mastertheblaster) [inline src/publish-scoped.js]: Just a thought: maybe you should rethrow error and stop the flow?
Cause now if connection issue will appear with artifactory, you will treat as package doesn't exist.
Though a build will succeed and we will have mismatched versions in artifactory and developer will not even know about this.

File: src/publish-scoped.js (hunk 1 above)
Comment (mastertheblaster) [reply src/publish-scoped.js]: Because lock files AFAIK had package name inside and they are published also.
So as a result the `shrinkwrap` file might refer to `@wix/package`, but published package name will be `package`.

Comment (Schniz) [reply src/publish-scoped.js]: done

File: src/publish-scoped.js (hunk 4 above)
Comment (Schniz) [reply src/publish-scoped.js]: The only problem I have with this flow is that I'd rather cancel the whole flow if that's the case. wdyt?
